specs/ingestion/common/schemas/transformation.yml

```
@@ -25,9 +30,49 @@ Transformation:
     - createdAt
     - updatedAt
 
+TransformationType:
+  type: string
+  description: The type of transformation, which can be either 'code' or 'noCode'.
+  enum:
+    - code
+    - noCode
+
+TransformationInput:
+  description: The input for the transformation, which can be either code or a no-code configuration.
+  oneOf:
+    - $ref: '#/TransformationCode'
+    - $ref: '#/TransformationNoCode'
+
+TransformationCode:
+  type: object
+  additionalProperties: false
+  description: Input for a transformation that contains the source code of the transformation.
+  properties:
+    code:
+      type: string
+      description: The source code of the transformation.
+  required:
+    - code
+
+TransformationNoCode:
+  type: object
+  additionalProperties: false
+  description: Input for a no-code transformation that contains a series of steps.
+  properties:
+    steps:
+      type: array
+      items:
+        $ref: '#/TransformationNoCodeStep'
+  required:
+    - steps
+
+TransformationNoCodeStep:
+  type: object
+  description: A step in a no-code transformation.
+
 Code:
   type: string
```

```suggestion
  type: string
  deprecated: true
```

I guess we can move it there now as well

---

sure, it is moved with https://github.com/algolia/api-clients-automation/pull/4901/commits/ea0503cc9d194645cadba27f6905cdca26296491